src/compiler/config/validate-config.ts

```diff
@@ -86,11 +84,10 @@ export const validateConfig = (
     hydratedFlag: validateHydrated(config),
     logger,
     outputTargets: config.outputTargets ?? [],
-    packageJsonFilePath: join(rootDir, 'package.json'),
-    rootDir,
     sys: config.sys ?? bootstrapConfig.sys ?? createSystem({ logger }),
     testing: config.testing ?? {},
     transformAliasedImportPaths: userConfig.transformAliasedImportPaths ?? false,
+    ...validatePaths(config),
```

Comment:
Do we still need the call to `validatePaths()` [here](https://github.com/ionic-team/stencil/blob/139f4cb81f3465b4baa2b92dc1b3d394519c7249/src/compiler/config/validate-config.ts#L167) now?

Reply:
@tanner-reits I think we do - IIUC, `validatePaths` returns the following properties we spread into the `ValidatedConfig`:
- `rootDir `
- `packageJsonFilePath`
------
- `srcDir`
- `cacheDir`
- `srcIndexHtml`
------
- `globalScript` (opt)
- `globalStyle` (opt)
- `buildLogFilePath` (opt)

Of that first section, we want there to be a source of truth for `rootDir` and `packageJsonPath`, which prior to this commit we were setting in this file:
- [rootDir](https://github.com/ionic-team/stencil/pull/4329/files#diff-53c120bef3205f89229fbe98a5aadfcb51012ead144729ade2fc597c1e9ad5e9L80)
- [packageJsonPath](https://github.com/ionic-team/stencil/pull/4329/files#diff-53c120bef3205f89229fbe98a5aadfcb51012ead144729ade2fc597c1e9ad5e9L89)
but are now setting in `validatePaths`:
- [rootDir](https://github.com/ionic-team/stencil/pull/4329/files#diff-acce362c35ee08b13791b7dd3b053c58903f2a0636efd3a3259dcb5f4a0837d9R29)
- [pacakgeJsonFilePath](https://github.com/ionic-team/stencil/pull/4329/files#diff-acce362c35ee08b13791b7dd3b053c58903f2a0636efd3a3259dcb5f4a0837d9R49)

That middle section of:
- `srcDir`
- `cacheDir`
- `srcIndexHtml`

Corresponds to the new non-optional properties we added to the `StrictConfigFields` type [here](https://github.com/ionic-team/stencil/pull/4329/files#diff-c396c2ecfcbdf4562ae518122990719c12b5ed9bd64908ef2142620f6d1440dbR448).

The latter three are optional on `StencilConfig`, but appears to be correctly setting them on the `ValidatedConfig` if it's set

Reply:
Bah - somehow I missed the link in @tanner-reits's question 🤦